### PR TITLE
PHPCS: Have the analyzer use PSR12

### PR DESCRIPTION
**Pull request by kraftbj**
Right now, the Analyzer was not built with WPCS in mind at all, but we still run it against our usual ruleset.

#### Changes proposed in this Pull Request:

* Excludes the analyzer from WPCS.
* Adds PSR 12 only for the analyzer.

#### Jetpack product discussion

n/a

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* CI happy? (It may complain about files being part of the exclude list if they are PSR-12 compliant. This PR does not attempt to change any package files.)

### CODE REVIEW

File: .phpcs.xml.dist [hunk 1]
```diff
@@ -5,7 +5,10 @@
 	<rule ref="./.phpcs.config.xml" />
 
 	<!-- Default ruleset. -->
-	<rule ref="Jetpack" />
+	<rule ref="Jetpack">
+		<!-- Ignore Analyzer. Does not follow WPCS -->
+    	<exclude-pattern>/projects/packages/analyzer/*</exclude-pattern>
```
Comment (anomiex) [inline .phpcs.xml.dist:10]: Fix indentation
```suggestion
		<exclude-pattern>/projects/packages/analyzer/*</exclude-pattern>
```

File: .phpcs.xml.dist [hunk 2]
```diff
@@ -40,4 +43,8 @@
 		<exclude name="Generic.PHP.Syntax" />
 	</rule>
 
+		<rule ref="PSR12">
+    		<include-pattern>/projects/packages/analyzer/*</include-pattern>
+    	</rule>
```
Comment (anomiex) [inline .phpcs.xml.dist:48]: Here too
```suggestion
	<rule ref="PSR12">
		<include-pattern>/projects/packages/analyzer/*</include-pattern>
	</rule>
```

**Review by anomiex — COMMENTED** [review]
> CI happy?

Nope. PSR12 seems to include a few things we probably don't want, such as PSR2.Classes.ClassDeclaration.OpenBraceNewLine and Generic.WhiteSpace.DisallowTabIndent.TabsUsed.

**Comment by kraftbj** [comment]
> Nope. PSR12 seems to include a few things we probably don't want, such as PSR2.Classes.ClassDeclaration.OpenBraceNewLine and Generic.WhiteSpace.DisallowTabIndent.TabsUsed.

That makes sense, thanks. I didn't think about the currently-compliant files in the package either. I'll take a fresh swing.

**Comment by kraftbj** [comment]
Going to block this until after https://github.com/Automattic/jetpack/pull/22035 lands.

Only way to land this without causing a lot of manual rebasing for 22035 is to add files to the exclude list, which we don't allow. In 22035, I'm exempting the analyzer from WPCS. Once it lands, I'll revisit to add PSR-12 (modified) and get everything compliant.

**Comment by jeherve** [comment]
Closing this for now because of the lack of activity on this. We can always reopen in the future if needed.

**Comment by kraftbj** [comment]
Thanks for cleaning this up. With the other advances in the monorepo (e.g. per-project phpcs rules), going about this others ways is better and already done.

**Pull request closed without merging**